libdnf/repo/solvable/Package.cpp

```diff
@@ -0,0 +1,235 @@
+#include "Package.hpp"
+#include "DependencyContainer.hpp"
+
+class libdnf::Package::Impl {
+public:
+    Impl() = default;
+    Impl(const Impl &source);
+private:
+    friend class libdnf::Package;
+
+    void createLibsolvable(HyRepo repo);
+    void fillSolvableData(const std::string &name, const std::string &version, const std::string &arch) const;
+    void fillSolvableData(const std::string &name, const std::string &version, const std::string &arch,
+                          const std::vector<Dependency> &dependencies) const;
+    std::shared_ptr<DependencyContainer> getDependencies(Id type, Id marker = -1) const;
+    Queue *getDependencyQueue(Id type, Id marker) const;
+
+    Id id;
+    DnfSack *sack;
+};
+
+libdnf::Package::Impl::Impl(const Impl &source)
+{
+    id  = source.id;
+    sack = source.sack;
+}
+
+void libdnf::Package::Impl::createLibsolvable(HyRepo repo)
```

Comment:
I thing this function should be part of fillSolvableData.

Reply:
No, `fillSolvableData` would do two things, which is not acceptable by me.